File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-3.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public int goodNodes(TreeNode root) {
        return dfs(root, -150);
    }
    private int dfs(TreeNode root, int high) {
        if(root == null) return 0;
        int cur = (root.val >= high) ? 1 : 0;
        high = Math.max(high, root.val);
        return cur + dfs(root.left, high) + dfs(root.right, high);
    }
}

File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-3.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        st = []
        cur = root

        while True:
            while cur:
                st.append(cur)
                cur = cur.left
            
            node = st.pop()

            k = k - 1
            if k == 0:
                return node.val
            
            cur = node.right